java/modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/utils/sslcert/ocsp/OCSPVerifier.java

```diff
@@ -220,7 +220,12 @@ private OCSPReq generateOCSPRequest(X509Certificate issuerCert, BigInteger seria
      *
      */
 
-    @SuppressWarnings({"startswith"}) //FALSE POSITIVE: URI is checked by calling gettagno
+    @SuppressWarnings({"startswith"})
+    //TRUE POSITIVE: String returned wouldn't always be a URI and hence we can't make a stub file
+    //for it but we know here it will be a URI because of the check-
+    //(genName.getTagNo() == GeneralName.uniformResourceIdentifier). More information can be found on-
+    //https://people.eecs.berkeley.edu/~jonah/bc/org/bouncycastle/asn1/x509/GeneralName.html#getTagNo()
+    //It is still not guaranteed that the url string will only have the accepted protocols.
```

Comment:
From this explanation, I still don't understand where the actual URIs being used (which seem to flow from the `AccessDescription`s) come from. Can you clarify?

A writing tip: the most important part of this explanation comes at the end, which is confusing. Try to put the most important facts (here, why this warning is a true positive) first, and then follow up with supporting details.